### PR TITLE
[New Term] C++ Math Functions: `fmax()`

### LINKED ISSUE #964: [Term Entry] C++ Math Functions: `fmax()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `fmax()` function.

- In the `terms/` folder, make a new folder called `fmax`
- Inside the `fmax` folder, create a new markdown file called `fmax.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by borntofrappe**
### Description

Closes #964 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by borntofrappe** [comment]
For the examples I thought of adding one where one of the value is `NaN`, for instance storing the square root of a negative number. Ultimately I thought it would be beyond the scope of the entry to use the `sqrt` function.

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 3;
  double y = sqrt(-2);
  double result;
  result = fmax(x, y);
  std::cout << "The larger value between " << x << " and " << y << " is " << result << "\n";
}
```

I'm more than open to suggestions.

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]:
```suggestion
The parameters `x` and `y` are usually a numeric [data type](https://www.codecademy.com/resources/docs/cpp/data-types). If one of the parameters is `NAN`, then the other parameter is returned.
```

We should also include something here about how this functions with pos/neg `INFINITY`.

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Hey @borntofrappe! This entry is off to a great start! 🚀 I left you a few comments/suggestions, but not that many. Let me know when you've finished implementing the feedback or if you have any questions. 😄

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (borntofrappe) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:27]: You are more than right...quite a silly mistake

Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:27]: No worries at all! Happens to all of us. 😄

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (borntofrappe) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]: Would something like the following work?

```text
Any numeric value is larger than negative `INFINITY` except negative `INFINITY` and `NAN`. Any numeric value is smaller than positive `INFINITY` except `INFINITY` and `NAN`.
```

As an alternative, and _before_ mentioning the specific instance where one of the parameter is `NAN`, I could add the following:

```text
Values are ordered from negative `INFINITY` to positive `INFINITY`.
```

In this manner one understands that `fmax` returns negative `INFINITY` if the other value is `NAN`.

Let me know if you have a preference or a better approach.

Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]: @borntofrappe What about something like this, instead?

```suggestion
The parameters `x` and `y` are usually a numeric [data type](https://www.codecademy.com/resources/docs/cpp/data-types) ranging from `-INFINITY` (negative) to `INFINITY` (positive). If one of the parameters is `NAN`, then the other parameter is returned.
```

I separated "negative" and "positive" into parentheses to preserve the C++ syntax of the two infinity values.

Comment (borntofrappe) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]: Excellent. I'll commit the change promptly.

**Review by Dusch4593 — APPROVED** [review]
@borntofrappe LGTM for the next review 👍🏻

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:3]:
```suggestion
Description: 'Returns the larger of two arguments.'
```

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:15]:
```suggestion
The **`fmax()`** function returns the larger of two arguments.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@borntofrappe, I did a second review and left a couple of comments.

**Review by SSwiniarski — APPROVED** [review]
@borntofrappe, @Dusch4593 Looks good to me.

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]: - Can we change `NAN` (all caps) to `NaN`
- The reference to positive `INFINITY` works, but perhaps we can change it to "from negative `INFINITY` to positive `INFINITY`."

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:38]: Perhaps we can add a period here before the new line (and edit the output to include it)

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:50]:
```suggestion
The following example returns the larger value between `-32.123` and `-32.231`:
```

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/fmax/fmax.md:61]:
```suggestion
  std::cout << "The larger value between " << x << " and " << y << " is " << result << "." << "\n";
```

**Review by yangc95 — COMMENTED** [review]
@borntofrappe I did a quick review, but this PR looks ready to merge after.

@Dusch4593 @SSwiniarski

File: content/cpp/concepts/math-functions/terms/fmax/fmax.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmax/fmax.md:23]: @yangc95 What about something like this? "from negative `-INFINITY` to positive `INFINITY`"?

**Comment by yangc95** [comment]
live link: https://www.codecademy.com/resources/docs/cpp/math-functions/fmax

Screen Shot 2022-10-23 at 1 37 23 PM

cc: @Dusch4593 @SSwiniarski

**Pull request merged**
